src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3623,17 +3623,29 @@ private BoundExpression BindNullCoalescingAssignmentOperator(AssignmentExpressio
                 return GenerateNullCoalescingAssignmentBadBinaryOpsError(node, leftOperand, rightOperand, diagnostics);
             }
 
-            // If the type of the left is Nullable<A0>, and the right is the default literal, the default will be
-            // converted to the default of Nullable<A0>, null, and not the default of A0. This is likely unintended
-            // on the part of the user, so we warn
-            if (leftType.IsNullableType() && rightOperand.IsLiteralDefault())
+            HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+            if (leftType.IsNullableType())
             {
-                Error(diagnostics, ErrorCode.WRN_DefaultLiteralConvertedToNullIsNotIntended, node.Right, leftType.GetNullableUnderlyingType());
+                // If B is implicitly convertible to A0, then the result type of this expression is A0, except if B is dynamic.
+                // This differs from most assignments such that you cannot directly replace a with (a ??= b).
+                var underlyingLeftType = leftType.GetNullableUnderlyingType();
+                var underlyingRightConversion = Conversions.ClassifyImplicitConversionFromExpression(rightOperand, underlyingLeftType, ref useSiteDiagnostics);
+                if (underlyingRightConversion.Exists && rightOperand.Type?.IsDynamic() != true)
+                {
+                    diagnostics.Add(node, useSiteDiagnostics);
+                    var convertedRightOperand = CreateConversion(rightOperand, underlyingRightConversion, underlyingLeftType, diagnostics);
+                    // Create a nullable conversion for use by the local rewriter
+                    var nullableConversion = Conversions.ClassifyImplicitConversionFromType(underlyingLeftType, leftType, ref useSiteDiagnostics);
```

Comment:
>nullableConversion [](start = 24, length = 18)

This conversion doesn't seem to be used (it's not saved into the bound node we create)

Reply:
Whoops, forgot to remove. I was using this at one point and then removed it from the bound node, but forgot to remove the creation.

---
In reply to: [293042697](https://github.com/dotnet/roslyn/pull/36329#discussion_r293042697) [](ancestors = 293042697)